flink-runtime/src/main/java/org/apache/flink/runtime/highavailability/AbstractHaServices.java

```diff
@@ -225,6 +237,13 @@ public CompletableFuture<Void> globalCleanupAsync(JobID jobID, Executor executor
                 executor);
     }
 
+    protected void cleanupClusterHaStoragePath() throws Exception {
+        final Path clusterHaStoragePath =
+                HighAvailabilityServicesUtils.getClusterHighAvailableStoragePath(configuration);
+        final FileSystem fileSystem = clusterHaStoragePath.getFileSystem();
+        fileSystem.delete(clusterHaStoragePath, true);
```

Comment:
Is this actually already delete the blob dir here? Because this will delete the parent path recursively. But the protocol of `BlobStoreService#closeAndCleanupAllData` maybe not limited to deleting directory, just the current implementation is. Maybe the call of `blobStoreService.closeAndCleanupAllData()` before remove the parent dir just looks a little confused but still needs to be kept.

Reply:
Yes, it will already delete the blob dir. However, as you mentioned, the protocol of BlobStoreService#closeAndCleanupAllData() is not limited to deleting directory, so it is still required.